packages/ai-jsx/src/react/completion.tsx

```diff
@@ -100,5 +80,35 @@ export async function* UICompletion(
     return <Component>{children.map(toComponent)}</Component>;
   }
 
-  return <AI.React>{toComponent(JSON.parse(modelResult))}</AI.React>;
+  const Element: z.Schema = z.object({
+    name: z.string(),
```

Comment:
I tried to restrict the `name` only to `reactComponents`, but my knowledge of `zod` failed me. Any thoughts @NickHeiner? Is it possible to do this if the list is created dynamically?